stdlib/@tests/test_cases/builtins/check_slice.py

```diff
@@ -0,0 +1,116 @@
+"""
+- Type hint `slice` should be compatible with the **"all-slices"**:
+    - `slice(None)`, `slice(None, None)` and `slice(None, None, None)`. (⟿ `slice[?, ?, ?]`)
+- Type hint `slice[T]` should be compatible with:
+    - `slice(None)`, `slice(None, None)` and `slice(None, None, None)` (⟿ `slice[?, ?, ?]`)
+    - `slice(t)`, `slice(None, t)` and `slice(None, t, None)`.  (⟿ `slice[?, T, ?]`)
+    - `slice(t, None)` and `slice(t, None, None)`.  (⟿ `slice[T, ?, ?]`)
+    - `slice(t, t)` and `slice(t, t, None)`.  (⟿ `slice[T, T, ?]`)
+- Type hint `slice[X, Y]` should be compatible with:
+    - `slice(None)`, `slice(None, None)` and `slice(None, None, None)` (⟿ `slice[?, ?, ?]`)
+    - `slice(y)`, `slice(None, y)` and `slice(None, y, None)`.  (⟿ `slice[?, Y, ?]`)
+    - `slice(x, None)` and `slice(x, None, None)` (⟿ `slice[X, ?, ?]`)
+    - `slice(x, y)` and `slice(x, y, None)`.  (⟿ `slice[X, Y, ?]`)
+-  Type hint `slice[X, Y, Z]` should be compatible with:
+    - `slice(None)`, `slice(None, None)` and `slice(None, None, None)`. (⟿ `slice[?, ?, ?]`)
+    - `slice(y)`, `slice(None, y)` and `slice(None, y, None)`.   (⟿ `slice[?, Y, ?]`)
+    - `slice(x, None)` and `slice(x, None, None)` (⟿ `slice[X, ?, ?]`)
+    - `slice(x, y)` and `slice(x, y, None)`.  (⟿ `slice[X, Y, ?]`)
+    - `slice(None, None, z)`  (⟿ `slice[?, ?, Z]`)
+    - `slice(None, y, z)`   (⟿ `slice[?, Y, Z]`)
+    - `slice(x, None, z)`   (⟿ `slice[X, ?, Z]`)
+    - `slice(x, y, z)`  (⟿ `slice[X, Y, Z]`)
+"""
+
+from typing import Any
+from typing_extensions import assert_type
+
+# region check slice constructors ------------------------------------------------------
+assert_type(slice(None), "slice[Any, Any, Any]")
+assert_type(slice(None, None), "slice[Any, Any, Any]")
+assert_type(slice(None, None, None), "slice[Any, Any, Any]")
+
+assert_type(slice(1), "slice[Any, int, Any]")
+assert_type(slice(None, 1), "slice[Any, int, Any]")
+assert_type(slice(None, 1, None), "slice[Any, int, Any]")
+
+assert_type(slice(1, None), "slice[int, Any, Any]")
+assert_type(slice(1, None, None), "slice[int, Any, Any]")
+
+assert_type(slice(1, 1), "slice[int, int, Any]")
+assert_type(slice(1, 1, None), "slice[int, int, Any]")
+
+assert_type(slice(1, 1, 1), "slice[int, int, int]")
+# endregion check slice constructors ---------------------------------------------------
```

Comment:
I like that you've separated the test file into sections, but I think a comment introducing each section is sufficient; not sure this one adds value

```suggestion
```

Reply:
Well, I use these because pycharm and vscode let you collapse such sections (https://code.visualstudio.com/docs/editor/codebasics#_folding)

![image](https://github.com/user-attachments/assets/c94f6d42-6d93-44ca-8348-e6f5ca818c4b)

Reply:
Oh, I see. I wasn't familiar with this convention. I'm still not a massive fan of this style, but I won't insist :-)